p2p/src/network.rs

```diff
@@ -344,14 +344,14 @@ impl<T: Pload, K: Kex, E: Enc> NetworkBase<T, K, E> {
         );
     }
 
-    fn disconnect_peer(&mut self, public_key: PublicKey) {
-        let peer = match self.peers.remove(&public_key) {
+    fn disconnect_peer(&mut self, public_key: &PublicKey) {
+        let peer = match self.peers.remove(public_key) {
             Some(peer) => peer,
             _ => return iroha_logger::warn!(?public_key, "Not found peer to disconnect"),
         };
         iroha_logger::debug!(listen_addr = %self.listen_addr, %peer.conn_id, "Disconnecting peer");
 
-        let peer_id = PeerId::new(peer.p2p_addr, public_key);
+        let peer_id = PeerId::new(peer.p2p_addr, public_key.clone());
```

Comment:
previous version had 0 `clone`s but now since you take public key by reference you have to clone

Reply:
Previous version of `PeerId::new` borrowed `PublicKey` and cloned it internally. I moved it outside.